runtime/connection_cache.go

```diff
@@ -25,6 +25,11 @@ var (
 	connCacheCloseLatencyMS = observability.Must(meter.Int64Histogram("connnection_cache.close_latency", metric.WithUnit("ms")))
 )
 
+type connectionCache struct {
+	conncache.Cache
+	hasHungConn bool
+}
```

Comment:
Could this be moved to a func on `conncache.Cache`? It could be called `HangingErr()` and return an error containing the name of the first connection that is hanging (cached from the hanging check). It would also enable auto-recover in case the connection actually stops hanging.